Intersect.Client.Core/Interface/Game/Spells/SpellItem.cs

```diff
@@ -1,247 +1,289 @@
-                    }
-                }
-                else
-                {
-                    if (Pnl.Texture != null)
-                    {
-                        Pnl.Texture = null;
-                    }
-                }
-
-                mTexLoaded = spell.Icon;
-                mCurrentSpellId = Globals.Me.Spells[mYindex].Id;
-                mIconCd = Globals.Me.IsSpellOnCooldown(mYindex);
-
-                if (mIconCd)
-                {
-                    mCooldownLabel.IsHidden = false;
-                    var remaining = Globals.Me.GetSpellRemainingCooldown(mYindex);
-                    mCooldownLabel.Text = TimeSpan.FromMilliseconds(remaining).WithSuffix();
-                }
+                _iconImage.Texture = spellTex;
+                _iconImage.RenderColor.A = (byte)(_cooldownLabel.IsVisibleInParent ? 100 : 255);
+                _iconImage.IsVisibleInParent = true;
             }
             else
             {
-                if (Pnl.Texture != null)
+                if (_iconImage.Texture != null)
                 {
-                    Pnl.Texture = null;
+                    _iconImage.Texture = null;
+                    _iconImage.IsVisibleInParent = false;
                 }
+            }
 
-                mTexLoaded = string.Empty;
+            if (_descWindow != null)
+            {
+                _descWindow.Dispose();
+                _descWindow = null;
+                _iconImage_HoverEnter(null, null);
             }
         }
 
         if (!IsDragging)
         {
-            if (mMouseOver)
+            if (_mouseOver)
             {
                 if (!Globals.InputManager.IsMouseButtonDown(MouseButton.Left))
                 {
-                    mCanDrag = true;
-                    mMouseX = -1;
-                    mMouseY = -1;
-                    if (Timing.Global.MillisecondsUtc < mClickTime)
+                    _canDrag = true;
+                    _mouseX = -1;
+                    _mouseY = -1;
+                    if (Timing.Global.MillisecondsUtc < _clickTime)
                     {
-                        mClickTime = 0;
+                        _clickTime = 0;
                     }
                 }
                 else
                 {
-                    if (mCanDrag && Draggable.Active == null)
+                    if (_canDrag && Draggable.Active == null)
                     {
-                        if (mMouseX == -1 || mMouseY == -1)
+                        if (_mouseX == -1 || _mouseY == -1)
                         {
-                            mMouseX = InputHandler.MousePosition.X - Pnl.ToCanvas(new Point(0, 0)).X;
-                            mMouseY = InputHandler.MousePosition.Y - Pnl.ToCanvas(new Point(0, 0)).Y;
+                            _mouseX = InputHandler.MousePosition.X - _iconImage.ToCanvas(new Point(0, 0)).X;
+                            _mouseY = InputHandler.MousePosition.Y - _iconImage.ToCanvas(new Point(0, 0)).Y;
```

Comment:
Calling `ToCanvas()` twice instead of caching the value is atrocious and should be fixed